### PR TITLE
Clustering active nodes

### PR DESCRIPTION
**Pull request by vkhatale**
[STRMALERT-1726]. Added API to get missing alerts and added unit test case for the same.

### CODE REVIEW

File: pkg/services/sqlstore/alert.go [hunk 1]
```diff
@@ -339,3 +342,47 @@ func GetAlertStatesForDashboard(query *m.GetAlertStatesForDashboardQuery) error
...
+	cmd := &m.GetAllAlertsQuery{}
+	if err := bus.Dispatch(cmd); err != nil {
+		sqlog.Error("Could not load alerts", "error", err)
+		return err
+	}
+
+	//Find Missing alerts
+	var evalTime int64
+	var frequency int64
+	missedAlerts := make([]*m.Alert, 0)
```
Comment (sanchitraizada) [inline pkg/services/sqlstore/alert.go:379]: There are two competing goals here
1. Allow the normal processing in each node to evaluate all alerts within 1 min interval (and thus not requiring missing alert runs)
2. Detect and evaluate any missing alert but not delay "next" normal processing on the node chosen for evaluating missing alert - It might so happen that the node chosen for evaluating missing alert wouldn't be able to participate in next normal run and thus, create more missing alerts.

Operationally speaking, normal processing might have higher significance (since it will be using latest data points), hence it might make sense to set a configurable upper bound on the length of the array for missing alerts.

File: pkg/services/sqlstore/alert.go [hunk 2]
```diff
@@ -339,3 +342,47 @@ func GetAlertStatesForDashboard(query *m.GetAlertStatesForDashboardQuery) error
...
+	for _, alert := range cmd.Result {
+		evalTime = alert.EvalDate.Unix()
+		frequency = alert.Frequency
+		if evalTime < ts-frequency && evalTime >= ts-2*frequency {
```
Comment (sanchitraizada) [inline pkg/services/sqlstore/alert.go]: The criteria seem generally ok but consider having some configuration around how far we want to go back (just as a guard-rail). 

Few of the unlikely scenarios -  
1. frequency could be hourly/daily (incorrectly?) does it make sense to go that far
2. there can be weird cases where for some reason, a run might take more than few (3?) minutes on all nodes (assuming freq =1 min)

**Comment by sanchitraizada** [comment]
LGTM - just some minor comments

**Review by sanchitraizada — COMMENTED** [review]
LGTM (if possible, please add unit tests for the edge cases ) thanks!
